CMakeLists.txt

```diff
@@ -46,20 +48,19 @@ if(VULKAN_FOUND)
                     "$ENV{VULKAN_SDK}/bin"
                     )
     endif()
-    if(DEFINED Vulkan_dxc_EXECUTABLE)
-        message("Found DirectX Shader Compiler under ${Vulkan_dxc_EXECUTABLE}")
-    elseif()
-        message("Couldn't find DirectX Shader Compiler executable, make sure it is present in Vulkan SDK or add it manually via Vulkan_dxc_EXECUTABLE env variable")
+    if(Vulkan_dxc_EXECUTABLE)
+        message(STATUS "Found DirectX Shader Compiler under ${Vulkan_dxc_EXECUTABLE}")
+    else()
+        message(WARNING "Couldn't find DirectX Shader Compiler executable, make sure it is present in Vulkan SDK or add it manually via Vulkan_dxc_EXECUTABLE env variable. HLSL shaders won't be compiled")
+        unset(Vulkan_dxc_EXECUTABLE CACHE)
     endif()
 else()
-    message("VULKAN_SDK path not found, HLSL shaders won't be compiled")
+    message(WARNING "VULKAN_SDK path not found, HLSL shaders won't be compiled")
```

Comment:
I don't think we should put out a warning in this case. Having DXC available is fully optional, hence why we added compiled .spv for all HLSL shaders.

Reply:
OK, that's fine.  Would you prefer this as a STATUS message, or no message at all?

On the topic of many changes at once, I realize the review problem but tried to select items in this PR that work together to achieve a specific outcome - that is a successful batch mode run across all samples on macOS without crashes or early exit prior to all samples completing.  As you can see this touches quite a few things.  However,  I did specifically leave out Xcode and iOS/iOS simulator-related changes for a follow-on PR.  I believe that one can be smaller and more focussed on a few areas.  In any case, please take your time and I am happy to provide explanations and/or make changes based on your feedback.

Reply:
> OK, that's fine. Would you prefer this as a STATUS message, or no message at all?

The latter would be fine. Not having DXC only affects maintainers, a very small group of people.

> On the topic of many changes at once, I realize the review problem but tried to select items in this PR that work together to achieve a specific outcome

And that's totally fine. Just wanted to mention that we often take quite some time to review larger PRs ;)

Reply:
Removed Vulkan_SDK not found WARNING message in commit https://github.com/KhronosGroup/Vulkan-Samples/pull/1084/commits/6342fc08678d634417068e21d005b4f1e91f235f below. Retained existing message (now STATUS) if Vulkan_SDK found but dxc executable not found.